staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -328,6 +338,25 @@ func (g *Cloud) ensureInternalFirewall(svc *v1.Service, fwName, fwDesc string, s
 	if err != nil && !isNotFound(err) {
 		return err
 	}
+	if legacyFwName != "" {
```

Comment:
We should probably have a TODO somewhere for removing the legacy logic after X releases?

Reply:
Added.